pkg/controller/csr_check.go

```diff
@@ -159,44 +162,190 @@ func authorizeCSR(
 		return false, nil
 	}
 
+	var approvalErrors []error
+
 	// Check for an existing serving cert from the node.  If found, use the
 	// renewal flow.  Any error connecting to the node, including validation of
 	// the presented cert against the current Kubelet CA, will result in
 	// fallback to the original flow relying on the machine-api.
 	//
 	// This is only supported if we were given a CA to verify against.
+	var servingCert *x509.Certificate
 	if ca != nil {
-		servingCert, err := getServingCert(c, nodeAsking, ca)
-		if err == nil && servingCert != nil {
-			klog.Infof("Found existing serving cert for %s", nodeAsking)
+		var err error
+		servingCert, err = getServingCert(c, nodeAsking, ca)
+		if err != nil {
+			klog.Infof("Failed to retrieve current serving cert: %v", err)
+		}
+	}
 
-			err := authorizeServingRenewal(nodeAsking, csr, servingCert, x509.VerifyOptions{Roots: ca})
+	x509VerificationOpts := x509.VerifyOptions{Roots: ca}
+	if servingCert != nil {
+		klog.Infof("Found existing serving cert for %s", nodeAsking)
 
-			// No error, the renewal is authorized.
-			if err == nil {
-				return true, nil
-			}
+		err := authorizeServingRenewal(nodeAsking, csr, servingCert, x509VerificationOpts)
 
-			klog.Infof("Could not use current serving cert for renewal: %v", err)
-			klog.Infof("Current SAN Values: %v, CSR SAN Values: %v",
-				certSANs(servingCert), csrSANs(csr))
+		// No error, the renewal is authorized.
+		if err == nil {
+			return true, nil
 		}
 
-		if err != nil {
-			klog.Infof("Failed to retrieve current serving cert: %v", err)
-		}
+		approvalErrors = append(approvalErrors, err)
+		klog.Infof("Could not use current serving cert for renewal: %v", err)
+		klog.Infof("Current SAN Values: %v, CSR SAN Values: %v",
+			certSANs(servingCert), csrSANs(csr))
 	}
 
 	// Fall back to the original machine-api based authorization scheme.
 	klog.Infof("Falling back to machine-api authorization for %s", nodeAsking)
+	if err := authorizeServingCertWithMachine(machines, req, nodeAsking, csr); err != nil {
```

Comment:
maybe wrap it with condition which check machines presence, and do not engage it if no machines there? 
smth like
`if len(machines) > 0 {
... Use Mapi
} else {
... check renewal with egress ips
}`
WDYT?